airflow/dag_processing/bundles/base.py

```diff
@@ -112,3 +125,22 @@ def view_url(self, version: str | None = None) -> str | None:
         :param version: Version to view
         :return: URL to view the bundle
         """
+
+    @contextmanager
+    def lock(self):
+        if self._locked:
+            yield
+            return
+
+        lock_dir_path = self._dag_bundle_root_storage_path / "locks"
+        lock_dir_path.mkdir(parents=True, exist_ok=True)
+        lock_file_path = lock_dir_path / f"{self.name}.lock"
+        with open(lock_file_path, "w") as lock_file:
```

Comment:
this should be opened with `r` IIRC

Reply:
The nice thing about `w` is I don't have to create the file ahead of time. But no strong opinion here.

Reply:
`r` is to allow concurrent writers, many things taking shared lock

Reply:
Also found this:

> On at least some systems, LOCK_EX can only be used if the file descriptor refers to a file opened for writing.

We really only want exclusive locks for this purpose too.